Next version

Keats opened the pull request.

Keats committed:
Next version

tranzystorekk committed:
Update arg parsing to clap v3 (#1717)

* Update arg parsing to clap v3

Keats committed:
Add stemmer for other languages in the docs

Close #1740
Close #1349

Keats committed:
Add link to theme in syntax doc

Close #1679

Keats committed:
Mention paginator in getting started overview

Close #1658

Keats committed:
Mention translations for default language

Close #1634

Keats committed:
Mention Sass version support

Close #1619

Keats committed:
Update deps

mwcz committed:
Fix markup for fenced code with linenos (#1746)

* fix fenced code blocks with linenos and minify_html

* fix test cases involving linenos

srstevenson committed:
Include path separator in log message (#1749)

This adds a path separator between the directory and filenames in the
message printed by `zola serve` to indicate it is listening for changes.

Before:

    Listening for changes in /path/to/dir{config.toml, content, sass, static, templates}

After:

    Listening for changes in /path/to/dir/{config.toml, content, sass, static, templates}
                                         ^

Keats committed:
Add libs subcrate (#1747)

* Add libs subcrate

* Remove last direct lazy_static usage

* Update pinned ci version

* Update deps

kartva committed:
Make ignored_content work with nested paths and directories (#1738)

* Make ignored_content work with directories
Just had to remove strip_prefix(path) from file ignore code.
Added tests for subdirectory globbing.

* Add documentation

* add more tests
to confim that simple filename globs still match paths
without strip_prefix

liushuyu committed:
build: include metadata in Windows binary (#1735)

mwcz committed:
ignore links to '#' (#1756)

aslowwriter committed:
fix crash in build.rs when installing from source on linux (#1760)

Keats committed:
Update netlify deploy docs

Keats committed:
Refactoring rendering tests

southerntofu committed:
Bugfix check links i18n (close #1694) (#1780)

* Treat _index.LANG.md as section, not page (close #1694)
Display more advanced information in case of a future crash here

* Add link to translated local section in test_site_i18n to catch #1694
It appears linking to another translated page does not trigger the bug,
but linking to a section within the same (translated) page does.

* Allow (and test for) homepage internal links in broken link detection

Co-authored-by: southerntofu <southerntofu@thunix.net>

aslowwriter committed:
Make sure that zola serve/build can start from anywhere inside dir tree (#1761)

* Make sure that zola serve/build can start from anywhere inside dir tree

* make clippy and rustfmt a bit happier

* replace unecessary if-else with unwrap_or and display which path could not get canonicalized if it fails at startup

* canonicalize config_path to avoid crash when config.toml changes

Keats committed:
Update pulldown_cmark

piedoom committed:
Added XML support for load_data (#1769)

Co-authored-by: doomy <2640792-_doomy@users.noreply.gitlab.com>

Keats committed:
Update deps

Keats committed:
Load extra themes even with theme=css

Closes #1775

mwcz commented on src/main.rs:
```diff
@@ -1,124 +1,93 @@
+    let root_dir = cli_dir
+        .ancestors()
+        .find_map(|a| if a.join(&cli.config).exists() { Some(a) } else { None })
+        .unwrap_or_else(|| panic!("could not find directory containing config file"));
```
@Keats I think something in here breaks `zola init`.  I tried init today on the next branch and it immediately panics with this error.

Keats replied:
Hm, I'll fix it but we should really have some test for the cli :(

Keats committed:
Fix zola init

mwcz committed:
switch from chrono to time (#1790)

* switch from chrono to time

* remove some unused time features

* get utc offset before threading begins

asimpletune committed:
Allow load_data to take a literal (#1794) (#1794)

* Add `literal` as a new entry for `data source`, to be used by the `load_data` function

* Add tests to the module for plain text, json, xml, toml, and csv

* Update error messaging to include literal as a potential choice

* Update site documentation to include instructions for using `load_data` with a literal

Keats committed:
Try to change windows image for CI (#1807)

Keats committed:
Update deps

Keats committed:
Use anyhow (#1816)

Also fixes #1783

luishfonseca committed:
rename extra_highligh_themes to extra_syntaxes_and_themes (#1828)

extra_highligh_themes does not exist

mwcz committed:
apply skip_prefixes before parsing external link domain (#1833)

* apply skip_prefixes before parsing external link domain

* log number of links skipped by skip_prefixes

Keats committed:
Re-organise content and remove slotmap indirection level (#1827)

Keats committed:
Remove duplicate code

Keats committed:
Print path on metadata call failure

Keats committed:
Some more refactoring

mwcz committed:
fix typo in `zola init` (#1836)

Keats committed:
Remove unused deps thanks to cargo-machete

Keats committed:
Allow taxonomies to not be rendered

Closes #1750

Keats committed:
Pass content_path to populate_library

elfsternberg committed:
Create pull request/add yaml to load data (#1842)

* Add YAML to formats supported by load_data()

A fairly trivial addition; JSON and YAML are handled so similarly
that this was a matter of copying the JSON-relevant handlers and
editing the copies to handle YAML as well.  The test file was
literally generated with 'json2yaml'.

The documentation has been updated to indicate that load_data() now
handles YAML code.

The CHANGELOG has been updated as well.

* After checking, I found that it's generally agreed the mime type is still application/x-yaml.

* Update comment, unify library importing.

I noticed one more place where the list of formats was supported,
and added YAML to that list.

I noticed that there's a singular place to load the `libs::` crate,
and unified by importing of serde_yaml in that place.

Keats committed:
Build taxonomies as pages are added (#1841)

Keats committed:
Serialize transparent field in section front-matter

Closes #1840

Keats committed:
Rename links utils mod

Keats committed:
More efficient sitemap building

Keats committed:
Avoid cloning components when rendering feed

Keats committed:
Clippy

Keats committed:
Update syntect

Keats committed:
Clone Zola Tera instance for Markdown filter

Keats committed:
Ignore section with render=false when looking for path collisions

Closes #1656

Keats committed:
Fix tests

Keats committed:
Expose backlinks

colindean committed:
Enables setting base-url to / for relative URLs in serve mode (#1853)

Having to change the base URL to whatever ngrok URL was provisioned for
me was frustrating. This patch enables setting it to `/`, which will
then make the `get_url()` function simply return `/`.

Keats committed:
Fix tests

mwcz committed:
add link_checker settings for external_level and internal_level (#1848)

* add external_level and internal_level

* remove unnecessary debug derive on LinkDef

* clarify doc comment about link check levels

* simplify link checker logging

* add missing warn prefix

* simplify link level logging, remove "Level" from linklevel variants

* remove link level config from test site

* switch back to using bail! from get_link_domain

* move console's deps to libs

* remove unnecessary reference

* calling console::error/warn directly

* emit one error, or one warning, per link checker run

* various link checker level changes

* add docs about link checker levels

* remove accidentally committed test site

* remove completed TODO

Keats committed:
DRY up the external link checking a bit

Keats committed:
Clippy

lnicola committed:
Add `rel="alternate"` to Atom post links (#1868)

This improves compatibility with some readers, e.g. Miniflux, which otherwise don't detect these.

Jieiku committed:
Your message (#1880)

mattico committed:
Update to elasticlunr-rs 3.0.0 (#1873)

bemyak committed:
Fix clippy lint warnings (#1888)

bemyak committed:
Fix taxonomy current path (#1882)

* Fix current_path and current_url for taxonomies

* Fix language duplication in get_url fn

bemyak committed:
Always follow symlinks (#1883)

Keats committed:
Update changelog and deps

Keats committed:
Reenable link rewriting for colocated assets

All credits go to https://github.com/getzola/zola/pull/1779

Keats committed:
Mention CSS files in static assets overview

Closes #1892

Keats committed:
Do not assume themes are present for CSS export

Closes #1884

Keats committed:
Set current_term earlier in search JS

Closes #1856

Keats committed:
Fix old docker image reference

Closes #1861

Keats committed:
Specify mime version

Closes #1845

Keats committed:
Mention PAT for GH deploys

Closes #1785

Keats committed:
Merge branch 'master' into next

Keats committed:
Remove simplify theme

Closes #1863

bemyak committed:
Make robots.txt compliant with RFC (#1895)

The [RFC](http://www.robotstxt.org/orig.html) mentions only `Disallow`
directive, so it must appear in the file.

`Allow` is an ad hoc agreement between search engines that no all of
them follow.

Keats committed:
Fix feed location for taxonomies not in the default language

Keats committed:
Bump version of Zola used on netlify

Keats committed:
woops

Keats committed:
Add title_bytes sorting method

nyanpasu64 committed:
Fix incorrect documentation for syntax directory (#1901)

Currently the Configuration docs says to load syntax files into
config.toml [markdown] extra_syntaxes. However, ever since commit
23064f57c8 (released in Zola v0.15.0), the extra_syntaxes property was
replaced by extra_syntaxes_and_themes, and used as both syntax and color
theme search paths. Following the docs and trying to set the
extra_syntaxes property does nothing, and #1723 ran into this issue.

Change the docs to consistently reference extra_syntaxes_and_themes.

phillord committed:
Allow new lines before anchor (#1905)

Previously the heuristic check for links required spaces before the
attribute to ensure that attributes suffixed with `id` were not
identified. This has now been expanded to any white space character to
enable the `id` attribute to start on a new line.

phillord committed:
Properly escape anchor names (#1908)

agrandville committed:
fix windows EOL in markdown files (#1911)

* add a test

* CodeBlock text may be split on multiple parts

The CodeBlock text events are now concatenated and processed in a single stream at the end. cf https://github.com/raphlinus/pulldown-cmark/issues/457

Keats committed:
Stop printing invalid files in dirs

Too many invalid stuff like emacs files

bemyak committed:
Apply orientation transformation based on EXIF data (#1912)

Keats committed:
Insert anchor as full heading (#1916)

* Add insert_anchor = "heading"

* Update CHANGELOG.md

Co-authored-by: Alice Cecile <alice.i.cecile@gmail.com>

Keats committed:
Update deps

Keats committed:
Update pinned rustc version

Keats committed:
Fix some bugs + cargo fmt

Keats committed:
Update changelog

Keats committed:
Ensure homepages get their translations filled

Keats committed:
Update deps
